tests/foreman/ui/test_host.py

```diff
@@ -626,8 +632,31 @@ def test_negative_delete_primary_interface(session, module_host_template):
     :CaseLevel: System
     """
     interface_id = gen_string('alpha')
+    host_template = entities.Host(organization=module_org, location=module_location)
+    host_template.create_missing()
+    interface = (
+        'type=interface,mac={},identifier=eth0,name={},domain_id={},'
+        'ip={},primary=true,provision=true'
+    ).format(host_template.mac, gen_string('alpha'), host_template.domain.id, gen_ipaddr())
+    os_name = f'{host_template.operatingsystem.name} {host_template.operatingsystem.major}'
```

Comment:
I don't think pushing all of this into the test scope is a good pattern for the helpers refactor.  We're trading one helper for another, and needlessly creating entities on the satellite.

There is a `module_host_template` fixture in use by this test, which with the UI helper was used to populate the operating system fields.

@jyejare The direction we should be going with test behaviors like this is to expand our fixture definitions to cover these use cases.  The CLI factory function isn't really necessary to use here. We can indirectly parametrize the existing `pytest_fixtures.component.host.function_host` in order to specify additional arguments, if necessary.

Nailgun's `Host.create_missing()` function handles many/all of the necessary fields for creating a fake managed host, and the test setup in robottelo should leverage that. If `interface` isn't being handled right now, the `nailgun.entities.Host` class should be updated.

There is a third module using the `ui.create_fake_host` method, which does not need to create the host via UI for test coverage and would also benefit from a globally defined fixture.